src/api/apps/articles/model/retrieve.js

```diff
@@ -0,0 +1,201 @@
+const _ = require("underscore")
+const { ObjectId } = require("mongojs")
+
+export const toQuery = function(input, callback) {
```

Comment:
Can use the es6 syntax here, 
`export const toQuery = (input, callback) => {`

Also this should be ok to use `.ts` rather than `.js`